src/Sulu/Bundle/MediaBundle/Command/FormatCacheRegenerateFormats.php

```diff
@@ -0,0 +1,115 @@
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\MediaBundle\Command;
+
+use Sulu\Bundle\MediaBundle\Media\FormatManager\FormatManagerInterface;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\Finder\Finder;
+use Symfony\Component\Finder\SplFileInfo;
+
+class FormatCacheRegenerateFormats extends Command
+{
+    protected static $defaultName = 'sulu:media:regenerate-formats';
+
+    /**
+     * @var Filesystem
+     */
+    private $fileSystem;
+
+    /**
+     * @var FormatManagerInterface
+     */
+    private $formatManager;
+
+    /**
+     * @var string
+     */
+    private $localFormatCachePath;
+
+    public function __construct(
+        Filesystem $filesystem,
+        FormatManagerInterface $formatManager,
+        string $localFormatCachePath
+    ) {
+        parent::__construct();
+
+        $this->fileSystem = $filesystem;
+        $this->formatManager = $formatManager;
+        $this->localFormatCachePath = $localFormatCachePath;
+    }
+
+    protected function configure()
+    {
+        $this->setDescription('Loops over sulu image cache, and regenerates the existing images')
+        ;
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $ui = new SymfonyStyle($input, $output);
+
+        $finder = new Finder();
+        $finder->in(\realpath($this->localFormatCachePath));
+        $files = $finder->files();
+
+        if (!\count($files)) {
+            $ui->writeln(\sprintf('No images to regenerate found in "%s".', $this->localFormatCachePath));
+
+            return 0;
+        }
+
+        $progressBar = $ui->createProgressBar(\count($files));
+
+        $ui->writeln('Starting to regenerate: ' . \count($files) . ' images');
+        $ui->writeln('');
+
+        /** @var SplFileInfo $file */
+        foreach ($files as $file) {
+            $fileInformation = $this->getFileInformationArrayFromPath($file->getRelativePathname());
+
+            $this->formatManager->returnImage(
+                $fileInformation['id'],
+                $fileInformation['formatKey'],
+                $fileInformation['fileName']
+            );
+
+            $progressBar->advance();
+        }
+
+        $progressBar->finish();
+
+        $ui->writeln('');
+        $ui->writeln('');
+
+        $ui->success(\sprintf('Finished regenerating of "%s" images.', \count($files)));
+
+        return 0;
+    }
+
+    private function getFileInformationArrayFromPath($path): array
+    {
+        $exploded = \explode('/', $path);
+
+        $directories = \explode('/', $path, 2);
+        $fileNameParts = \explode('-', $directories[1], 2);
+
+        return [
+            'id' => $fileNameParts[0],
```

Comment:
Didn't got time yet to test it but this looks incorrect for me.

If I interpret the code correctly you are giving the relative path to it which looks like the following:

```php
$path = '50x50/01/245-test.jpg;
```

The result should be in this case:

```php
[
     'id' => 245,
     'formatKey' => '50x50',
     'fileName' => 'test.jpg',
]
```

But for me it looks like in your case 'id' is returned as `01/245`:

Think the correct code would look like the following:

```php
$pathParts = \explode('/', $path);
$formatKey = \reset($pathParts);
$filenameParts = explode('-', \end($pathParts), 2);
$id = (int) $filenameParts[0];
$fileName = $filenameParts[1];
```

We should add a Unit test create a test directory with different cases and check there if the FormatManager is called correctly.

Test cases which should work:

 - `50x50/01/1-test.jpg` 
 - `50x50/1-test.jpg`  // no segment
 - `50x50-inset/02/1-test-test.jpg` // filename and format name with `-`